File: Lab2/Lab2 C#/Armor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2_C_
{
    abstract class Armor
    {
        public string Name { get; }
        public double Weight { get; }
        public double Price { get; }

        protected Armor(string name, double weight, double price)
        {
            Name = name;
            Weight = weight;
            Price = price;
        }
        public abstract void Equip();
    }
}

File: Lab2/Lab2 Java/src/Armor.java
abstract class Armor {
    private String name;
    private double weight;
    private double price;

    public Armor(String name, double weight, double price) {
        this.name = name;
        this.weight = weight;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public double getWeight() {
        return weight;
    }

    public double getPrice() {
        return price;
    }

    // Abstract method to be implemented by concrete armor types
    public abstract void equip();
}
